src/lib/hesiod.c

```diff
@@ -141,6 +140,12 @@ int hesiod_init(void **context)
 	      else
 		errno = ENOMEM;
 	    }
+	  /* Make sure that the rhs is set. */
+	  if (!ctx->rhs)
+	    {
+	      errno = ENOEXEC;
```

Comment:
It's copied from read_config_file(), though I have no idea why read_config_file() uses that particular error code.  Would ENOENT make more sense?